### PR TITLE
Platform fees

### PR DESCRIPTION
**Pull request by kelvinlau20100**
associated sdk changes for PR https://github.com/zetamarkets/zeta-options/pull/322

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -234,6 +243,24 @@ export class Client {
       client._pendingUpdate = true;
     }
 
+    let [whitelistTradingFeesAddress, _whitelistTradingFeesNonce] =
+      await utils.getUserWhitelistTradingFeesAccount(
+        Exchange.programId,
+        wallet.publicKey
+      );
+    client._whitelistTradingFeesAddress = whitelistTradingFeesAddress;
+
```
Comment (xjcaa) [inline src/client.ts:252]: I think what you can do here is just - 

1. Get PDA
2. Check existence
3. If exist, set whitelistTradingFeesAddress to pda, else leave as undefined and add log, else dont log about whitelist for non whitelisting.

Comment (xjcaa) [reply src/client.ts:252]: Then you dont need the trading fees boolean.

File: src/client.ts [hunk 2]
```diff
@@ -374,15 +401,29 @@ export class Client {
...
-      marketIndex,
-      price,
-      size,
-      side,
-      this.marginAccountAddress,
-      this.publicKey,
-      openOrdersPda
-    );
+    let orderIx;
+    if (this._whitelistedTradingFees) {
```
Comment (xjcaa) [inline src/client.ts:405]: Then this can be - 
```
if (this._whitelistedTradingFees !== undefined) ...
```

File: src/client.ts [hunk 3]
```diff
@@ -374,15 +401,29 @@ export class Client {
...
+      orderIx = await placeOrderIx(
+        marketIndex,
+        price,
+        size,
+        side,
+        this.marginAccountAddress,
+        this.publicKey,
+        openOrdersPda,
+        this._whitelistTradingFeesAddress
```
Comment (xjcaa) [inline src/client.ts:414]: If `this._whitelistTradingFeesAddress` == undefined, you could just directly pass it in without needing an if else and then check if it's undefined in the program-instruction `placeOrderIx` function instead and it'd be cleaner.

File: src/client.ts (hunk 1 above)
Comment (kelvinlau20100) [reply src/client.ts:252]: yeah this makes a lot of sense

**Review by xjcaa — APPROVED** [review]

**Pull request merged**